### PR TITLE
Remote management, secret bundling, README updates

### PR DESCRIPTION
**Pull request by mbland**
Far from complete, but hopefully will get everyone a lot closer to getting started.

cc: @jcscottiii @ccostino @afeld @catherinedevlin @batemapf @ertzeid @mtorres253 @andrewmaier @jbarnicle

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,47 +1,172 @@
...
+   script](./go) command line interface requires version 2.3.0 or greater. You
+   may wish to first install a version manager such as
+   [rbenv](https://github.com/rbenv/rbenv) to manage and install different
+   Ruby versions.
 
-```
+1. Install [Docker](https://www.docker.com/) on your system. The commands
+   encapsulated in the [`./go` script](./go) are based on version 1.10.0.
+
+1. Run `docker-machine` start to start the Docker host, followed by
```
Comment (jcscottiii) [inline README.md]: you ended your single quote too soon.
`docker-machine start`

Comment (mbland) [inline README.md]: Fixed!

File: README.md [hunk 2]
```diff
@@ -1,47 +1,172 @@
...
+   build_images` within your copy of the repository.
+
+## Deployment
+
+1. Install your public SSH key on the remote host machine.
+
+1. Set the `REMOTE_HOST` and `REMOTE_ROOT` variables in the `./go` script as
+   necessary.
+
+1. Run `./go init_remote` if running on a brand new server. Otherwise run
```
Comment (afeld) [inline README.md:42]: Your plan is to use [ECS](https://aws.amazon.com/ecs/), right?

File: README.md [hunk 3]
```diff
@@ -1,47 +1,172 @@
...
+   # Testing locally with 18F/knowledge-sharing-toolkit
+   # Run `docker-machine env` to get the current IP.
+   192.168.99.100 pages.18f.gov
+   192.168.99.100 pages-staging.18f.gov
+   192.168.99.100 pages-internal.18f.gov
+   192.168.99.100 pages-releases.18f.gov
+   192.168.99.100 team-api.18f.gov
+   192.168.99.100 hub.18f.gov
+   192.168.99.100 handbook.18f.gov
+   ```
```
Comment (afeld) [inline README.md:75]: Won't this be a problem when you're trying to access the live sites? Not sure that we should recommend this.

File: go [hunk 4]
```diff
@@ -224,7 +239,7 @@ end
 
 command_group :cleanup, 'Image and container cleanup commands'
 
-def_command :rm_containers, 'Remove stopped (non-data) containers' do |args|
+def_command :rm_containers, 'Remove stopped non-data containers' do |args = []|
```
Comment (afeld) [inline go]: Uhh, I don't think you can have defaults for block parameters in Ruby. Just tried this in IRB:

``` ruby
> [1,2,nil,3].each do |el = 'foo'|
*  puts el.inspect
* end
1
2
nil
3
```

File: README.md (hunk 3 above)
Comment (mbland) [inline README.md:75]: Well, folks have to remember to delete or to comment them back out is all.

File: README.md (hunk 2 above)
Comment (mbland) [inline README.md:42]: Maybe, if I have time to figure it out this week. For the moment, I'm shooting on at least making the dedicated EC2 instance easy to grok. (But I may have time for ECS, just don't know right now.)

File: go [hunk 5]
```diff
@@ -80,6 +81,20 @@ DAEMONS = {
...
+SECRETS_BUNDLE_FILE = "#{SECRETS_BUNDLE_NAME}.tar.bz2"
+SECRET_FILES = %w(
+  */config/env-secret.sh
+  nginx/config/auth/pages-passwords.txt
+  nginx/config/ssl/dhparam*
+  nginx/config/ssl/keys/*
+  pages/config/pages.secret
+  ssh/config/id_rsa*
+  ssh/config/known_hosts*
+)
```
Comment (afeld) [inline go:96]: The different types of config files here seem very inconsistent...can we move everything to be configured the same way? Fine if you want to break this out to a separate Issue.

File: README.md (hunk 2 above)
Comment (afeld) [inline README.md:42]: This is a larger discussion, but this just means these systems have become a _new_ special snowflake (albeit a better documented/scripted one) relative to what we had before...does that make sense?

File: go (hunk 4 above)
Comment (mbland) [inline go]: Huh. Wonder where I got that from. Removed.

File: go (hunk 5 above)
Comment (mbland) [inline go:96]: Nginx and SSH are special cases; the one Pages thing could be moved from a file to an env var, given an update to pages-server. The rest are using this `*/config/env-secret.sh` convention. File an issue if you like.

File: README.md (hunk 2 above)
Comment (mbland) [inline README.md:42]: Well, it's a melting snowflake, I hope. By making it easier to bring the system up on a new machine via the `./go` script, running containerized services, I hope the next step is finding better ways to provision it.

File: go (hunk 5 above)
Comment (afeld) [inline go:96]: https://github.com/18F/knowledge-sharing-toolkit/issues/12

File: README.md (hunk 3 above)
Comment (afeld) [inline README.md:75]: Sure. I'm still hesitant to suggest it, but I'd at least make a note of that in the README (for those that may not understand `/etc/hosts`).

Comment (mbland) [inline README.md:75]: But...this is the README...and I do say that right above the example here. :-)

Comment (afeld) [inline README.md:75]: I just mean that I could see someone (like myself) following these directions to the letter, then a week later trying to access hub.18f.gov and being VERY confused about why it's not working. I'm going for principle of least surprise here.

**Pull request merged**